### PR TITLE
[PyCDE] Moving the most opinionated python out of bindings (#1179)

### LINKED ISSUE #1179: [Py] Move `design_entry` and anything opinionated out of "bindings"
We need to figure out a line between syntactic sugar on the bindings and "opinionated language"-y things.

Not sure where this should move yet, but it should be in the CIRCT repo just for practicality reasons -- about 1/2 the changes we make to the language-y stuff require changes to either the python bindings proper or the CIRCT C++ code.

### PR DESCRIPTION
**Pull request by teqdruid**
- Creating new frontends top level directory which contains frontends which are   optionally included.
- Moving @module and @generator into PyCDE frontend.
- Keeping `connect` in the bindings since it's used in the bindings tests.
- Fixing the integration tests.

Closes #1179.

### CODE REVIEW

**Comment by lattner** [comment]
Nice, thank you for driving this!

File: frontends/PyCDE/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  ${CMAKE_CURRENT_SOURCE_DIR}/lit.cfg.py
+)
+
+add_lit_testsuite(check-pycde "Running the PyCDE tests"
+  ${CMAKE_CURRENT_BINARY_DIR}
+  DEPENDS ${PyCDE_TEST_DEPENDS}
+  )
+set_target_properties(check-pycde PROPERTIES FOLDER "PyCDETests")
+
+# add_lit_testsuites(PyCDE ${CMAKE_CURRENT_SOURCE_DIR}
```
Comment (mikeurbach) [inline frontends/PyCDE/test/CMakeLists.txt:35]: Is this commented out from debugging or is the intent to uncomment at some point? Seems like there is a lit testsuite for check-pycde.

**Review by mikeurbach — APPROVED** [review]
Looks good to me, thanks for driving the split. I guess as a follow up we can set up a github workflow that enables the frontend and runs the test, perhaps nightly like the integration tests?

File: frontends/PyCDE/test/CMakeLists.txt (hunk 1 above)
Comment (teqdruid) [reply frontends/PyCDE/test/CMakeLists.txt]: Debugging. It's in `test/CMakeLists.txt`, so I copied it here but it was causing issues. I don't know what it does, but `--target check-pycde` works fine, so...

**Comment by teqdruid** [comment]
@mikeurbach thanks for the review!

**Pull request merged**